src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,95 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K,V>[] table;
+    private int threshold;
+    private int size;
+
+    public MyHashMap() {
+        this.table = (Node<K,V>[]) new Node[DEFAULT_INITIAL_CAPACITY];
+        this.threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        int index = getIndex(key);
+        Node<K,V> inputNode = new Node<>(key, value);
+        Node<K,V> currentNode = table[index];
+
+        if (currentNode == null) {
+            table[index] = inputNode;
+        }
 
+        while (currentNode != null) {
+            Node<K,V> nextNode = currentNode.next;
+            if (currentNode.key == key || key != null && key.equals(currentNode.key)) {
+                currentNode.value = value;
+                return;
+            } else if (nextNode == null) {
+                currentNode.next = inputNode;
+            }
+            currentNode = nextNode;
+        }
+
+        if (++size > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        final int index = getIndex(key);
+        Node<K,V> currentNode = table[index];
+
+        while (currentNode != null) {
+            if (currentNode.key == key || key != null && key.equals(currentNode.key)) {
+                return currentNode.value;
+            } else {
+                currentNode = currentNode.next;
+            }
```

Comment:
```suggestion
            } 
            currentNode = currentNode.next;
```
We do not need **else** after **return**